transport.go

```diff
@@ -0,0 +1,409 @@
+package quic
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/quic-go/quic-go/internal/wire"
+
+	"github.com/quic-go/quic-go/internal/protocol"
+	"github.com/quic-go/quic-go/internal/utils"
+	"github.com/quic-go/quic-go/logging"
+)
+
+type Transport struct {
+	// A single net.PacketConn can only be handled by one Transport.
+	// Bad things will happen if passed to multiple Transports.
+	//
+	// If the connection satisfies the OOBCapablePacketConn interface
+	// (as a net.UDPConn does), ECN and packet info support will be enabled.
+	// In this case, optimized syscalls might be used, skipping the
+	// ReadFrom and WriteTo calls to read / write packets.
+	Conn net.PacketConn
+
+	// The length of the connection ID in bytes.
+	// It can be 0, or any value between 4 and 18.
+	// If unset, a 4 byte connection ID will be used.
+	ConnectionIDLength int
```

Comment:
Maybe omit `ConnectionIDLength` entirely in favor of `ConnectionIDGenerator`?

Reply:
I consider `ConnectionIDGenerator` somewhat of a fringe use case. You only need it if you want to use a Connection ID-based load balancer.